magma/backend/coreir/coreir_transformer.py

```diff
@@ -31,10 +33,31 @@
 _logger = root_logger().getChild("coreir_backend")
 
 
+_generator_callbacks = {}
+
+
 def _is_generator(ckt_or_inst):
     return ckt_or_inst.coreir_genargs is not None
 
 
+def _coreir_longname(magma_defn_or_decl, coreir_module_or_generator):
+    # NOTE(rsetaluri): This is a proxy to exposing a pycoreir/coreir-c API to
+    # get a module's longname. This logic should be identical right now. Another
+    # caveat is that we don't elaborate the CoreIR generator at the magma level,
+    # so it's longname needs to be dynamically reconstructed anyway.
```

Comment:
We may consider adding an API for this, so internally coreir calls the same code to create the longname and we don't have to maintain two versions of the code.

Reply:
Yup agreed. See:

https://github.com/leonardt/pycoreir/issues/154

https://github.com/rdaly525/coreir/issues/1003